src/test/java/FileUtilityTest.java

```diff
@@ -243,4 +283,553 @@ public void testSetPermissions() throws IOException {
+                StandardWatchEventKinds.ENTRY_DELETE,
+                StandardWatchEventKinds.ENTRY_MODIFY,
+                StandardWatchEventKinds.OVERFLOW
+        );
+
+        Function<Path, Queue<WatchEvent.Kind<?>>> expectedEventsOracle = null;
+        switch (action) {
+            case "FOLDER_CREATE" -> {
+                expectedEventsOracle = path -> {
+                    Queue<WatchEvent.Kind<?>> expectedEvents = new LinkedList<>();
+                    expectedEvents.add(StandardWatchEventKinds.ENTRY_CREATE);
+                    if (!Files.isDirectory(path)) {
+                        expectedEvents.add(StandardWatchEventKinds.ENTRY_MODIFY);
+                    }
+                    return expectedEvents;
+                };
+                expectedPathToEventKindMap = getExpectedEvents(pathToRegister, expectedEventsOracle, maxDepth);
+                createDirectoriesRecursively(pathToRegister, 1_000);
+            }
+            case "FILE_CREATE" -> {
+                expectedEventsOracle = path -> {
+                    Queue<WatchEvent.Kind<?>> expectedEvents = new LinkedList<>();
+                    expectedEvents.add(StandardWatchEventKinds.ENTRY_CREATE);
+                    if (!Files.isDirectory(path)) {
+                        expectedEvents.add(StandardWatchEventKinds.ENTRY_MODIFY);
+                    }
+                    return expectedEvents;
+                };
+                expectedPathToEventKindMap = getExpectedEvents(pathToRegister, expectedEventsOracle, maxDepth);
+                createDirectoriesRecursively(pathToRegister.getParent(), 1_000);
+                Files.createFile(pathToRegister);
+            }
+            case "FOLDER_MODIFY", "FILE_MODIFY" -> {
+                expectedEventsOracle = path -> {
+                    Queue<WatchEvent.Kind<?>> expectedEvents = new LinkedList<>();
+                    expectedEvents.add(StandardWatchEventKinds.ENTRY_MODIFY);
+                    return expectedEvents;
+                };
+                expectedPathToEventKindMap = getExpectedEvents(pathToRegister, expectedEventsOracle, maxDepth);
+                Files.setLastModifiedTime(pathToRegister, FileTime.from(Instant.now()));
+            }
+            case "FOLDER_DELETE", "FILE_DELETE" -> {
+                expectedEventsOracle = path -> {
+                    Queue<WatchEvent.Kind<?>> expectedEvents = new LinkedList<>();
+                    expectedEvents.add(StandardWatchEventKinds.ENTRY_DELETE);
+                    return expectedEvents;
+                };
+                expectedPathToEventKindMap = getExpectedEvents(pathToRegister, expectedEventsOracle, maxDepth);
+                FileUtility.deleteRecursively(pathToRegister);
+            }
+            default -> throw new IllegalArgumentException("Invalid Action - " + action);
+        }
+        logger.info("Sleeping for 1 second and waiting for file changes events to be generated.");
```

Comment:
Internationalize the logged string.
Don't hardcode the timeout value, refer from variable.